llama/generation.py

```diff
@@ -71,6 +72,10 @@ def generate(
         tokens = tokens.to(device)
         input_text_mask = tokens != self.tokenizer.pad_id
 
+        temperature_tensor = torch.tensor(temperature).to(device)
```

Comment:
nit, can you leave a comment explaining why we need to convert them to tensors?

Reply:
Added